packages/driver/src/cy/snapshots.ts

```diff
@@ -10,6 +10,90 @@ export const HIGHLIGHT_ATTR = 'data-cypress-el'
 
 export const FINAL_SNAPSHOT_NAME = 'final state'
 
+type SelectorNode = {
+  frameId?: string
+  selector: string
+  ownerDoc: Document | ShadowRoot
+  host?: SelectorNode
+}
+
+const returnShadowRootIfShadowDomNode = (node: Element): ShadowRoot | null => {
```

Comment:
Does `node.shadowRoot` not work?

Reply:
I think I just tried to get something working and didn't revisit it 😳 . That should work or we can see if the root is an instance of shadow root

Reply:
@ryanthemanuel OK I think I remember a bit on this decision. `shadowRoot` only exists on the DOM node OUTSIDE the current shadowDOM (i.e. the `host`). We could check the `host` property instead but I' not sure if there are other cases where `host` might be set. We could also do something like `node.parentNode.host.shadowRoot` but honestly might be safest to check the serialized string.

also `instanceof` doesn't work with ShadowRoot in this case. So doing `node.parentNode.host.shadowRoot instanceof ShadowRoot` returns `false`.

Reply:
I wonder if its worth documenting this in the method here. I'll add something along with the changelog

Reply:
It still seems like it would be more straightforward to use `host`/`shadowRoot`.

Reply:
I think we can do the `node.host.shadowRoot` and verify that property exists which is likely pretty safe. We can't just check the `host` property mainly because its popular, kind of like in the case of anchor elements https://developer.mozilla.org/en-US/docs/Web/API/HTMLAnchorElement/host. I'll refactor this to use `node.host.shadowRoot`

Reply:
opted for `node.host.shadowRoot` in 9132ea1225